phantom-jdk8/src/main/scala/com/outworkers/phantom/jdk8/DefaultJava8Primitives.scala

```diff
@@ -37,27 +37,12 @@ trait DefaultJava8Primitives {
     }
   }
 
-  implicit object ZonedDateTimeIsPrimitive extends Primitive[ZonedDateTime] {
+  implicit val zonePrimitive: Primitive[ZoneId] = Primitive.derive[ZoneId, String](_.getId)(ZoneId.of)
 
-    override type PrimitiveType = java.util.Date
-
-    val cassandraType = CQLSyntax.Types.Timestamp
-
-    override def asCql(value: ZonedDateTime): String = {
-      value.toInstant.toEpochMilli.toString
-    }
-
-    override def fromRow(column: String, row: GettableByNameData): Try[ZonedDateTime] = nullCheck(column, row) {
-      r => ZonedDateTime.ofInstant(r.getTimestamp(column).toInstant, ZoneOffset.UTC)
+  implicit val zonedDateTimePrimitive: Primitive[ZonedDateTime] = {
+    Primitive.derive[ZonedDateTime, (Long, String)](dt => dt.toInstant.toEpochMilli -> dt.getZone.getId) {
```

Comment:
@mmatloka Yes basically, which is why I was going to add the backwards incompatible notice. The problem is you cannot recover the original `ZonedDateTime` back unless the information is somehow persisted.

Do you have other suggestions?